### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by Chike1990**
In this Milestone, I:
created a method called current_user in the ApplicationController to make current user data available to all controllers.
created forms to perform the following functions:
     creates a Post on behalf of the current_user;
     create a comment on behalf of the current_user;
     and allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/likes_controller.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      @like.author_id = @user.id
+      @like.post_id = @post.id
+      @like.save
+    end
+    redirect_to user_post_path(@user.id, @post.id)
+  end
+
+  def liked?
+    Like.where(author_id: @user.id, post_id: @post.id).exists?
+  end
```
Comment (andersonlebon) [inline app/controllers/likes_controller.rb:18]: - [ ] KIndly solve this error that occurs when a post is liked 
image

File: app/views/comments/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<div class="form-container">
+  <h1>Create New Comment</h1>
+  <%= form_with model: [@user, @post, @comment] do |f| %>
+    <h3>Text:</h3>
+    <%= f.text_area :text, placeholder: "Write your text...", required: true, class: "text-area" %>
+    <div class="submit-button">
+      <%= f.submit "Submit", class: "button submit" %>
+    </div>
+  <% end %>
+</div>
```
Comment (andersonlebon) [inline app/views/comments/new.html.erb:10]: - [ ] Kindly make sure this error is solved when a comment is created, check the redirection when a comment is created.

image

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @Chike1990 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
## Required Changes ♻️

- [ ] Kindly check comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by danielufeli — APPROVED** [review]
Hi @Chike1990 ,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

a

## Highlights
- Nice commits messages✔️
- Descriptive PR ✔️
- Good use of GitHub flow ✔️
- Clean Code ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @danielufeli in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**